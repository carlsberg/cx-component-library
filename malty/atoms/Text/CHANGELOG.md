# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2024-05-31)

### Bug Fixes

- **DSM-457:** after animation message is not displayed correctly on safari ([#407](https://github.com/CarlsbergGBS/cx-component-library/issues/407)) ([29f7f19](https://github.com/CarlsbergGBS/cx-component-library/commit/29f7f19ce9aecff9228df79ad47c90fa53121284))
- **DSM-520:** updated prop as on Text ([#533](https://github.com/CarlsbergGBS/cx-component-library/issues/533)) ([d425b73](https://github.com/CarlsbergGBS/cx-component-library/commit/d425b734ae45d3807f5528f82c817ec2af599b0c))
- Remove typography provider from every component ([#383](https://github.com/CarlsbergGBS/cx-component-library/issues/383)) ([5d05680](https://github.com/CarlsbergGBS/cx-component-library/commit/5d05680e40d3da5bb3042bd8191427a70606afc3))
- reverted changes ([#396](https://github.com/CarlsbergGBS/cx-component-library/issues/396)) ([ecfcf57](https://github.com/CarlsbergGBS/cx-component-library/commit/ecfcf573863253fea16e24764b6f452c3cc4f215))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- **alertInline:** [DSM-730] improve word breaking ([#755](https://github.com/CarlsbergGBS/cx-component-library/issues/755)) ([a938911](https://github.com/CarlsbergGBS/cx-component-library/commit/a9389112782d5603b457dae46ae798b36f72c91f))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-228:** Added rating component ([#379](https://github.com/CarlsbergGBS/cx-component-library/issues/379)) ([ab47354](https://github.com/CarlsbergGBS/cx-component-library/commit/ab47354db803fb051a1f87d66ee34966698df8bd))
- **DSM-22:** Toast ([#176](https://github.com/CarlsbergGBS/cx-component-library/issues/176)) ([94faba6](https://github.com/CarlsbergGBS/cx-component-library/commit/94faba6eb4324c4975ae16dac0ddb1611cc18ea7))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **DSM-439:** Change font-family behavior ([#470](https://github.com/CarlsbergGBS/cx-component-library/issues/470)) ([ab6dad4](https://github.com/CarlsbergGBS/cx-component-library/commit/ab6dad414c42c4343de3de6abaa4cd3ff98cfca1))
- **DSM-455:** Add new custom props to text component ([#412](https://github.com/CarlsbergGBS/cx-component-library/issues/412)) ([e88d260](https://github.com/CarlsbergGBS/cx-component-library/commit/e88d2601cd1223226ea5f5c0db17dbb29fef00f5))
- **DSM-515:** product card ([#548](https://github.com/CarlsbergGBS/cx-component-library/issues/548)) ([9b91827](https://github.com/CarlsbergGBS/cx-component-library/commit/9b918271c05d4a98fdd7c5b34bb356f89a92c991))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- integrate visual testing ([#925](https://github.com/CarlsbergGBS/cx-component-library/issues/925)) ([35dfc1c](https://github.com/CarlsbergGBS/cx-component-library/commit/35dfc1cce6b76d5d2661d552f9fb25477d3a57f7))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- **text:** [DSM-835] add tests ([#807](https://github.com/CarlsbergGBS/cx-component-library/issues/807)) ([45b1676](https://github.com/CarlsbergGBS/cx-component-library/commit/45b1676ade4527a0e89c536eabb6ee1e8955c5af))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
