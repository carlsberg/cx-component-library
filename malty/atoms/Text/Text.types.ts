export interface TextProps extends React.HTMLAttributes<HTMLParagraphElement> {
  textStyle: TextStyle;
  align?: TextAlign;
  color?: TextColor;
  italic?: boolean;
  as?: React.ElementType;
  ellipsis?: boolean;
  width?: string;
  children: string | JSX.Element;
  className?: string;
  dataQaId?: string;
}

export enum TextStyle {
  MediumBold = 'medium_bold',
  MediumBoldUnderline = 'medium_bold--underline',
  MediumDefault = 'medium_default',
  MediumDefaultUnderline = 'medium_default--underline',
  MediumSmallBold = 'medium-small_bold',
  MediumSmallBoldUnderline = 'medium-small_bold--underline',
  MediumSmallDefault = 'medium-small_default',
  MediumSmallDefaultUnderline = 'medium-small_default--underline',
  SmallBold = 'small_bold',
  SmallBoldUnderline = 'small_bold--underline',
  SmallDefault = 'small_default',
  SmallDefaultUnderline = 'small_default--underline',
  TinyBold = 'tiny_bold',
  TinyDefault = 'tiny_default',
  MicroBold = 'micro_bold',
  MicroDefault = 'micro_default'
}
export enum TextAlign {
  Left = 'left',
  Center = 'center',
  Right = 'right'
}
export enum TextColor {
  DigitalBlack = 'digital-black',
  White = 'white',
  Support20 = '20',
  Support40 = '40',
  Support60 = '60',
  Support80 = '80',
  Support100 = '100',
  AlertLight = 'alert-light',
  AlertStrong = 'alert-strong',
  DisableLightTheme = 'disable-light-theme',
  DisableDarkTheme = 'disable-dark-theme',
  Fail = 'fail',
  FailLight = 'fail-light',
  NotificationLight = 'notification-light',
  NotificationStrong = 'notification-strong',
  Success = 'success',
  SuccessLight = 'success-light'
}
