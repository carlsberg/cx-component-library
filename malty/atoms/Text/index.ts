export { Text } from './Text';
export type { TextProps } from './Text.types';
export { StyledParagraph } from './Text.styled';
