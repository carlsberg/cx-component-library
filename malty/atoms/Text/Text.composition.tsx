import { MaltyThemeProvider } from '@carlsberg/malty.theme.malty-theme-provider';
import React from 'react';
import { Text } from './Text';

export const BasicText = () => (
  <MaltyThemeProvider theme="global">
    <Text content="this is a text block" />
  </MaltyThemeProvider>
);
