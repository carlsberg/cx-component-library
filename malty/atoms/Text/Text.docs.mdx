---
labels: ['react', 'typescript', 'text']
description: 'A Text component.'
---
import { MaltyThemeProvider } from '@carlsberg/malty.theme.malty-theme-provider';
import { Text } from './Text';

Text example:

A simple Text component with some text

// live component playground - if you remove the word live it turns into a code snippet
```js live
  <MaltyThemeProvider theme="global">
    <Text content="click me" />
  </MaltyThemeProvider>
```
