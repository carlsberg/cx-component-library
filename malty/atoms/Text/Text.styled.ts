import styled from 'styled-components';
import { Align, Color, Size, Weight } from './Text.types';

export const StyledParagraph = styled.p<{ size?: Size; weight?: string; align?: string; color?: Color }>`
  font-family: ${({ theme }) => theme.font.fontFamily.text};
  color: ${({ color, theme }) => (color ? theme.color[color].value : theme.color.default.value)};
  text-align: ${({ align }) => align ?? Align.Left};
  font-size: ${({ size, theme }) => (size ? theme.font.fontSizes[size] : theme.font.fontSizes.medium)};
  font-weight: ${({ weight }) => weight || Weight.Normal};
`;
