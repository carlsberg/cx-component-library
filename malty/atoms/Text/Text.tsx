import { globalTheme as defaultTheme } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledParagraph } from './Text.styled';
import { TextAlign, TextColor, TextProps, TextStyle } from './Text.types';

export const Text = ({
  textStyle = TextStyle.MediumDefault,
  align = TextAlign.Left,
  color = TextColor.DigitalBlack,
  italic = false,
  as = 'p',
  ellipsis = false,
  children,
  width,
  className,
  dataQaId,
  ...props
}: TextProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  let StyledTag = as;
  const allowedTags = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span'];

  if (as && allowedTags.includes(`${as}`)) {
    StyledTag = as;
  }

  return (
    <StyledParagraph
      data-testid={dataQaId}
      textStyle={textStyle}
      color={color}
      align={align}
      italic={italic}
      ellipsis={ellipsis}
      width={width}
      theme={theme}
      as={StyledTag}
      className={className}
      {...props}
    >
      {children}
    </StyledParagraph>
  );
};
