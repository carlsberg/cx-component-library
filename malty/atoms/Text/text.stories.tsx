import { Story } from '@storybook/react';
import React from 'react';
import { Text as TextComponent } from './Text';
import { Align, Color, Size, TextProps, Weight } from './Text.types';

export default {
  title: 'Atoms/Text',
  component: TextComponent,
  parameters: {
    importObject: 'Text',
    importPath: '@carlsberggroup/malty.atoms.text'
  },
  argTypes: {
    color: {
      options: Object.values(Color),
      description: 'Color of the headline, from the following options',
      control: {
        type: 'select'
      },
      table: {
        defaultValue: {
          summary: 'primary'
        }
      }
    },
    size: {
      options: Object.values(Size),
      description: 'Size of the headline, from the following options',
      control: {
        type: 'select'
      },
      table: {
        defaultValue: {
          summary: 'medium'
        }
      }
    },
    align: {
      options: Object.values(Align),
      description: 'Headline alignment, from the following options',
      control: {
        type: 'select'
      },
      table: {
        defaultValue: {
          summary: 'left'
        }
      }
    },
    children: {
      description: 'This is the content of a headline component',
      control: { type: 'text' }
    },
    weight: {
      options: Object.values(Weight),
      description: 'Weight of the typography.',
      table: { defaultValue: { summary: 'regular' } },
      control: {
        type: 'select'
      }
    },
    underline: {
      description: 'Should headline be underline?',
      table: { defaultValue: { summary: false } },
      control: { type: 'boolean' }
    },
    italic: {
      description: 'Should headline be italic?',
      table: { defaultValue: { summary: false } },
      control: { type: 'boolean' }
    }
  }
};
const Template: Story<TextProps> = ({ size, weight, align, color, children, underline, italic }) => (
  <TextComponent size={size} weight={weight} align={align} color={color} underline={underline} italic={italic}>
    {children}
  </TextComponent>
);

export const Text = Template.bind({});

Text.args = {
  size: Size.Small,
  children:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lorem augue, cursus ac sem in, fringilla sagittis ligula. Curabitur viverra laoreet convallis. Nam mi tortor, pellentesque sollicitudin pretium in, lacinia ut nunc.'
};
