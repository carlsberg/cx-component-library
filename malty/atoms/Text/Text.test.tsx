import { jsonRenderer, render, screen } from '@carlsberggroup/malty.utils.test';
import React from 'react';
import { TextStyle } from '.';
import { Text } from './Text';

const text = 'this is a text block';
describe('text', () => {
  it('matches snapshot', () => {
    const view = jsonRenderer(<Text textStyle={TextStyle.MediumBold}>{text}</Text>);
    expect(view).toMatchSnapshot();
  });

  it('should render with the correct text', () => {
    render(<Text textStyle={TextStyle.MediumBold}>{text}</Text>);
    const rendered = screen.getByText(text);
    expect(rendered).toBeInTheDocument();
  });
});
