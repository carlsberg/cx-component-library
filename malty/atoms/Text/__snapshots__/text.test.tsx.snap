// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text matches snapshot 1`] = `
<span
  className="sc-bdfBQB dGZTMR"
>
  <p
    className="sc-gsTEea dirViS"
  >
    this is a text block
  </p>
</span>
`;